R/mcmc.R

```diff
@@ -83,6 +77,10 @@ fit_mcmc <- function(
         optimizer = c("L-BFGS-B", "BFGS")
     )
 
+    if (mmrm_initial$failed) {
+        stop("Fitting MMRM to original dataset failed")
+    }
+
```

Comment:
What happened if it failed before this ?  Would it just carry on processing ?

Reply:
It was giving a less interpretable error when operating with the regression coefficients which where NA probably: 

`Error in stan_data$R %*% mmrm_initial$beta : requires numeric/complex matrix/vector arguments`

Reply:
ouch , yer good to have this here now !

Could you include a test for a model which can't converge and show that this catches it though please.  I know its trivial but its good to have encase we edit the code in the future and break this for some reason

Reply:
Ok, I will do it!

Reply:
Done!